### PR TITLE
Use Vector with locally relevant dofs in compute_stress

### PR DESCRIPTION
**Pull request by masterleinad**
Copying the parallel layout from `displacement`, we are missing some locally relevant DoFs we are accessing in `compute_stress`.

### CODE REVIEW

**Comment by Rombur** [comment]
```
An error occurred in line <1479> of file </scratch/source/dealii/include/deal.II/lac/la_parallel_vector.templates.h> in function
    dealii::LinearAlgebra::distributed::Vector<Number, MemorySpace>& dealii::LinearAlgebra::distributed::Vector<Number, MemorySpace>::operator-=(const dealii::LinearAlgebra::VectorSpaceVector<Number>&) [with Number = double; MemorySpace = dealii::MemorySpace::Host]
The violated condition was: 
    ::dealii::deal_II_exceptions::internals::compare_for_equality(locally_owned_size(), v.locally_owned_size())
Additional information: 
    Two sizes or dimensions were supposed to be equal, but aren't. They
    are 336 and 0.

Stacktrace:
-----------
#0  /opt/dealii/lib/libdeal_II.g.so.9.5.1: dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::Host>::operator-=(dealii::LinearAlgebra::VectorSpaceVector<double> const&)
#1  /var/jenkins/workspace/adamantine_PR-303/build/bin/test_mechanical_physics: adamantine::MechanicalPhysics<3, 4, adamantine::SolidLiquidPowder, dealii::MemorySpace::Host>::solve()
#2  /var/jenkins/workspace/adamantine_PR-303/build/bin/test_mechanical_physics: elastostatic::test_method()
```

**Comment by masterleinad** [comment]
Retest this please.

File: source/MechanicalPhysics.cc [hunk 1]
```diff
@@ -278,13 +278,19 @@ MechanicalPhysics<dim, p_order, MaterialStates, MemorySpaceType>::solve()
...
   // compute the plastic deformation.
-  auto incremental_displacement = displacement;
+  dealii::IndexSet locally_relevant_dofs =
+      dealii::DoFTools::extract_locally_relevant_dofs(_dof_handler);
+  dealii::LA::distributed::Vector<double, dealii::MemorySpace::Host>
+      incremental_displacement(
+          _dof_handler.locally_owned_dofs(), locally_relevant_dofs,
+          _mechanical_operator->rhs().get_mpi_communicator());
+  incremental_displacement = displacement;
   // TODO Remove this once we support refinement/material addition
```
Comment (Rombur) [inline source/MechanicalPhysics.cc:288]: You can remove the TODO. This PR fixes it.

**Comment by masterleinad** [comment]
Retest this please.

**Review by Rombur — APPROVED** [review]

**Pull request merged**